linux-kernel: add patches for USB on Loongson

KexyBiscuit opened the pull request:
Topic Description
-----------------

- linux-kernel: \(loongson3\) adjust config
    - Disable USB Onboard Device \(Hub\) which is known to cause boot issues on
    x86 and LoongArch devices, such as MacBook Pro 11,2. Disable it here \(as
    we have no reasons to enable it here either\).
    - Disable OTG support \(pointless\) and default external hub disablement,
    which would render Apple keyboards with USB passthrough unusable.
    Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: re-enable CPU_HWMON for Loongson 3
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: synchronize patches from linux-kernel
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: apply another workaround for Loongson 3 thermal issue
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 0986025d0d63e78bf3ebacbb20936fa16ae9abdb.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: build Loongson 3 platform device drivers with CPU_HWMON only
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 7191898c02d290f2268dc5852057166eb7db9b42.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: add patch to fix MTU for stmmac on Loongson
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 432f486752b7c9970538749878fbaf51c5c0b24a.
    - Update config for Loongson 3.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: synchronize patches from linux-kernel
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: updates for Loongson...
    - Add patch to fix S3 on LoongArch.
    - Original author claimed the patch as superseded, however no follow-up was found.
    - Disable CPU_HWMON and ARCH_SUPPORTS_HUGETLBFS for Loongson 3.
    - Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 5e0ce2d6d62c230e4e54d144f4bf8a4f705f3401.
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- linux-kernel: \(loongson3\) update config
    - \*\*FIXME\*\* Disable CPU_HWMON \(Loongson 3A/B CPU hardware monitoring\), as
    it is known to cause thermal shutdowns from erratic temperature
    readings - we observed this on Excelsior E23L board \(3B4000\) boards. We
    should re-enable this once we find the root cause to this behaviour.
    - \*\*FIXME\*\* Disable HugeTLB support as it causes TLB exceptions and
    illegal memory access on 3A4000 and 3B4000 \(possibly more, but we
    haven't tested other platforms yet\).
    - Enable as many drivers as reasonably expected for this platform \(and
    being very optimistic - as usual - about what users may connect or use
    with this platform\).
    - Move networking and complex storage/fiber \(with the exception of common
    LSI HBAs\) out of vmlinuz to save on precious RAM available whilst at the
    bootloader \(some firmware may have trouble swallowing boot payloads
    larger than \~64MiB\).
    Co-authored-by: Mingcong Bai <jeffbai@aosc.io>
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
- kernel-tools: synchronize patches from linux-kernel
    Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

... and 1 more commits

Package(s) Affected
-------------------

- kernel-tools: 6.13.0-10
- linux+kernel: 3:6.13.0
- linux-kernel-${__VER}: 1:6.13.0-11

Security Update?
----------------

No

Build Order
-----------

```
#buildit linux-kernel linux+kernel kernel-tools
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [x] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

KexyBiscuit committed:
linux-kernel: add patches for USB on Loongson...

- Update patch to fix full-/low-speed USB devices on ASUS XC-LS3A6M boards.
- Add patch to fix remote wakeup feature.
- Add patch to fix HCCPARAMS register error for LS7A EHCI.
- Track Loongson patches from Huacai Chen at https://github.com/AOSC-Tracking/chenhuacai-linux @ aosc/v6.13, current HEAD is 4af37903009f80487a17180d0d2397fde9b38f77.
- Track deepin patches at https://github.com/AOSC-Tracking/deepin-community-kernel @ aosc/v6.6, current HEAD is f65a1ce63e2b15af8fb76cdb725bef676e2bf1f9.
- Track deepin rolling patches at https://github.com/AOSC-Tracking/deepin-community-kernel-rolling @ aosc/v6.12, current HEAD is d40949c20fadb3a0e0a1269c1b0d6bd7dc5c513c.
- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 796424b4107f1277b2e76a4f4d3a566b9a042019.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: synchronize patches from linux-kernel

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: (loongson3) update config

- **FIXME** Disable CPU_HWMON (Loongson 3A/B CPU hardware monitoring), as
  it is known to cause thermal shutdowns from erratic temperature
  readings - we observed this on Excelsior E23L board (3B4000) boards. We
  should re-enable this once we find the root cause to this behaviour.
- **FIXME** Disable HugeTLB support as it causes TLB exceptions and
  illegal memory access on 3A4000 and 3B4000 (possibly more, but we
  haven't tested other platforms yet).
- Enable as many drivers as reasonably expected for this platform (and
  being very optimistic - as usual - about what users may connect or use
  with this platform).
- Move networking and complex storage/fiber (with the exception of common
  LSI HBAs) out of vmlinuz to save on precious RAM available whilst at the
  bootloader (some firmware may have trouble swallowing boot payloads
  larger than ~64MiB).

Co-authored-by: Mingcong Bai <jeffbai@aosc.io>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: updates for Loongson...

- Add patch to fix S3 on LoongArch.
  - Original author claimed the patch as superseded, however no follow-up was found.
- Disable CPU_HWMON and ARCH_SUPPORTS_HUGETLBFS for Loongson 3.
- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 5e0ce2d6d62c230e4e54d144f4bf8a4f705f3401.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: synchronize patches from linux-kernel

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: add patch to fix MTU for stmmac on Loongson

- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 432f486752b7c9970538749878fbaf51c5c0b24a.
- Update config for Loongson 3.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: build Loongson 3 platform device drivers with CPU_HWMON only

- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 7191898c02d290f2268dc5852057166eb7db9b42.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: apply another workaround for Loongson 3 thermal issue

- Track AOSC OS patches at https://github.com/AOSC-Tracking/linux @ aosc/v6.13, current HEAD is 0986025d0d63e78bf3ebacbb20936fa16ae9abdb.

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
kernel-tools: synchronize patches from linux-kernel

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

KexyBiscuit committed:
linux-kernel: re-enable CPU_HWMON for Loongson 3

Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai commented:
LoongArch:

Tested good on XA61200 with AMD Radeon R7 240 and the integrated LG110 graphics - in both cases, keyboards can be used to wake up the computer from S3, and wakeup completes successfully.

MingcongBai commented:
MIPS Loongson 3 tested good on the following:

- Lemote A1901/1906
- Excelsior E23L
- BM6J74

MingcongBai committed:
linux-kernel: (loongson3) adjust config

- Disable USB Onboard Device (Hub) which is known to cause boot issues on
  x86 and LoongArch devices, such as MacBook Pro 11,2. Disable it here (as
  we have no reasons to enable it here either).
- Disable OTG support (pointless) and default external hub disablement,
  which would render Apple keyboards with USB passthrough unusable.

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>

MingcongBai reviewed (approved):
Ship it!
